sway-types/src/source_engine.rs

```diff
@@ -91,8 +95,18 @@ impl SourceEngine {
         source_id
     }
 
-    pub fn get_autogenerated_source_id(&self, program_id: ProgramId) -> SourceId {
-        self.get_source_id_with_program_id(&Self::AUTOGENERATED_PATH.into(), program_id)
+    /// Return the associated autogenerated pseudo file for the passed `source_id.
+    /// Example: main.autogenerated.sw for main.sw
+    pub fn get_associated_autogenerated_source_id(&self, source_id: &SourceId) -> SourceId {
+        let path = self.get_path(source_id);
+        let file_name = PathBuf::from_str(path.file_name().unwrap().to_str().unwrap()).unwrap();
+        let path = path.with_file_name(format!(
+            "{}.{}.{}",
+            file_name.file_stem().unwrap().to_str().unwrap(),
+            Self::AUTOGENERATED_PATH,
+            file_name.extension().unwrap().to_str().unwrap()
```

Comment:
assuming these unwraps will not throw errors

Reply:
That is a good point. I am making this function fallible by returning an `Option`.